### DISCUSSION TITLE
JSONDecodeError when querying Grafana Mimir Endpoint

### DISCUSSION THREAD

**Discussion by mrjaved788** [google/slo-generator#500] [Q&A]
Hi Community Team,
We are planning to implement SLO Generator for hundreds of services within our organization. We have our application metrics in Grafana Mimir Database (which also uses PromQL to query the data and is similar to Prometheus).

So for POC, we are performing Error Rate calculations and we are getting following JSON response from our Mimir endpoint in the browser and Postman:
```
{
    "status": "success",
    "data": {
        "resultType": "vector",
        "result": [
            {
                "metric": {},
                "value": [
                    1722278818,
                    "0"
                ]
            }
        ]
    }
}
```

When we run the SLO Generator for the same query using Prometheus as Backend for Mimir Endpoint (as both use PromQL to query data) we get the following error:
**Command:**
slo-generator compute -f C:\Users\user1\Desktop\slo_generator\samples\prometheus\config_1.yaml -c C:\Users\user1\Desktop\slo_generator\samples\config.yaml --export
**Output:**
ERROR - prom-metrics-availability        | 5 mins   | JSONDecodeError: Extra data: line 1 column 5 (char 4) | Backend returned None.
ERROR - prom-metrics-availability        | 10 mins  | JSONDecodeError: Extra data: line 1 column 5 (char 4) | Backend returned None.
INFO - Run finished successfully in 2.4s.
INFO - Run summary | SLO Configs: 1 | Duration: 2.4s

So, what can be the possible reason for it:
1. JSON response coming in is invalid. If yes, then what should the format of data that is accepted by SLO Gen for calculations and how to refine the json response for SLO?
2. Mimir endpoint is not compatible with Prometheus as Backend for SLO Gen and we need custom backend to support Mimir.
3. Or anything else that we must look into.

Looking forward to a positive response
Thanks

**Comment by lvaylet** [comment]
Hi @mrjaved788, thanks a lot for all these details.

I am on vacation until the beginning of next week. I will take a look next week.

I am not an expert with Grafana Mimir. I will look at the documentation to understand how requests and queries are handled under the hood, and confirm whether the Prometheus backend can indeed be reused. I will also check if a Grafana Mimir client exists for Python, which would greatly simplify the creation of a custom backend.

**Reply by lvaylet** [reply]
Also, can you run the same command with debug mode enabled, so we get more details to work on?

```sh
DEBUG=1 slo-generator compute -f C:\Users\user1\Desktop\slo_generator\samples\prometheus\config_1.yaml -c C:\Users\user1\Desktop\slo_generator\samples\config.yaml --export
```

Note: This works on Linux. Not sure about Windows. You just need to figure out how to set an environment variable.

**Reply by mrjaved788** [reply]
The use of "DEBUG" helped me resolve my issue. I was able to use Mimir query endpoint with Prometheus as Backend(eliminating the need for a custom Backend). 
Thanks for the help. Appreciate it
